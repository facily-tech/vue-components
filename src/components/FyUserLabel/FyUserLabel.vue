<template>
  <div class="d-flex align-center justify-center">
    <fy-avatar v-bind.sync="$props" v-bind="$attrs">
      <slot />
    </fy-avatar>

    <div class="flex-column justify-space-between ml-5">
      <div class="font-weight-bold">{{ label }}</div>
      <div class="grey--text text--lighten-1">{{ caption }}</div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { VAvatar } from 'vuetify/lib';

const BaseVAvatar = Vue.extend({ mixins: [VAvatar] });

interface options extends InstanceType<typeof BaseVAvatar> {
  label?: string;
  caption?: string;
}

export default BaseVAvatar.extend<options>().extend({
  name: 'fy-user-label',

  components: { FyAvatar: () => import('@/components/FyAvatar') },

  props: {
    label: {
      type: String,
      default: '',
    },
    caption: {
      type: String,
      default: '',
    },
  },
});
</script>
