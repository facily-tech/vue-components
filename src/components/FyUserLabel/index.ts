export { default as FyUserLabel } from './FyUserLabel.vue';
