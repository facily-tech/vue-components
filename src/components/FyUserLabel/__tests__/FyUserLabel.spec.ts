import Vue from 'vue';
import Vuetify from 'vuetify';
import FyUserLabel from '../FyUserLabel.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyUserLabel.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyUserLabel>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyUserLabel, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
