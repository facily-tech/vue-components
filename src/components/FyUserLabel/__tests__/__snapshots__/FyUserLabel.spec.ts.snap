// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyUserLabel.ts should render component and match snapshot 1`] = `
<div class="d-flex align-center justify-center">
  <!---->
  <div class="flex-column justify-space-between ml-5">
    <div class="font-weight-bold"></div>
    <div class="grey--text text--lighten-1"></div>
  </div>
</div>
`;
